### ISSUE TITLE
Office365 autodiscover blocked by CORS policy in Outlook add-in

### ISSUE THREAD

**Issue by tomderoo** [OfficeDev/office-js#4797]
# Office365 autodiscover blocked by CORS policy in Outlook add-in

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Office on the web / Windows "new Outlook"
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: web / unknown
* Operating System: Linux / Windows
* Browser (if using Office on the web): Chrome (Firefox does NOT have the issue), and supposedly also Edge inside the "new Outlook client"
 
## Expected behavior
Our custom Outlook add-in is configured to be allowed to run on "own" as well as shared mailboxes.
When the add-in starts, `OfficeRuntime.auth.getAccessToken()` is started and concludes successfully, for "own" and shared mailboxes.
The add-in is allowed to properly access "own" and shared mailboxes.

## Current behavior

When the add-in starts, `OfficeRuntime.auth.getAccessToken()` is started and concludes successfully, BUT immediately afterward some Office process attempts a request to `https://outlook.office365.com/autodiscover/autodiscover.json/v1.0/my.shared.email@my.server?Protocol=REST` (with `my.shared.email@my.server` obviously replaced by the correct e-mail address of the shared mailbox), which then results in a CORS error on Chrome (and presumably Edge inside "new Outlook" - I cannot know for sure, since the client that reported the issue could not open inspector screen):
```
Access to XMLHttpRequest at 'https://outlook.office365.com/autodiscover/autodiscover.json/v1.0/my.shared.email@my.server?Protocol=REST' from origin 'https://outlook.office.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

I am unsure as to what Office JS process actually fires the autodiscover. I only know that it happens shortly after a seemingly successful resolution of `OfficeRuntime.auth.getAccessToken()`; that process properly retrieves an access token that I can confirm to be legit and as expected.

It also appears a _recent issue_; it started appearing "more or less a week ago" (as reported by the client that reported the issue to us). Everything worked fine before that. There were no changes made to the add-in code itself.

## Steps to reproduce

I do not know, since I do not know what fires the autodiscovery.

## Link to live example(s)

I have no live example. The add-in is currently sideloaded in beta and requires an account with a private backend for relevant functionality to appear. I can try to create a standalone example if the abovementioned error description gives you absolute no idea what causes it, but I will only do this if really necessary as I am severely pressed for time.

# Provide additional details

None that I can think of.

## Context

Add-in no longer works on shared mailboxes, where it used to work fine.

## Useful logs

Console errors from chrome:
```
Access to XMLHttpRequest at 'https://outlook.office365.com/autodiscover/autodiscover.json/v1.0/my.shared.email@my.server?Protocol=REST' from origin 'https://outlook.office.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```
(Replace `my.shared.email@my.server` with actual e-mail)
```
GET https://outlook.office365.com/autodiscover/autodiscover.json/v1.0/my.shared.email@my.server?Protocol=REST net::ERR_FAILED 401 (Unauthorized)
s @ owa.38196.m.b133693d.js:1
nc @ owa.38196.m.b133693d.js:1
re @ owa.38196.m.b133693d.js:1
(anonymous) @ owa.38196.m.b133693d.js:1
e.invoke @ owa.65283.m.7d3e3dd8.js:10
u @ owa.65283.m.7d3e3dd8.js:10
c @ owa.65283.m.7d3e3dd8.js:10
setInterval
v @ owa.65283.m.7d3e3dd8.js:10
```
(Replace `my.shared.email@my.server` with actual e-mail)

The piece of code that the error refers to is:
```js
if (l.isRemoteItem()) {
                m && (d.owner = u.mailboxSmtpAddress),
                o = d.owner;
                try {
                    s = await (i = d.owner,
                    new Promise((e,t)=>{
                        if (i) {
                            let n = new XMLHttpRequest;
                            n.open("GET", "https://outlook.office365.com/autodiscover/autodiscover.json/v1.0/" + i + "?Protocol=REST"),
                            n.onload = function() {
                                200 === n.status ? e(JSON.parse(this.response).Url) : t(Error("Request failed with status " + n.status + ". Message: " + n.responseText))
                            }
                            ,
                            n.send() // <--- Error happens here
```

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by anjalitp** [comment]
This is a duplicate of issue reported in https://github.com/OfficeDev/office-js/issues/4781 . Please track the same for more updates.

**Comment by tomderoo** [comment]
Thanks for your response @anjalitp , will check the other issue.
